### hokuyo_process_node lidar_data_processing
sobel_kernal_size: 3   # 1 or 3 is best
sobel_thres_value: 600 # once kernal size changed it will also change
min_line_point_num: 10 # at least 10 points in a line
box_left_right: 80    # distance to left and right (0, 120)
box_top_bottom: 200    # distance to front (0, 239) 240
lidar_img_height: 250  # box_left_right *2 < lidar_img_height
lidar_img_width: 250   # box_top_bottom < lidar_img_width
min_x_dis: 1.5         # tolerance of x direction of lidar
ignore_num: 5          # tolerance of y direction lo lidar

### visual_controller line_tracking and block_servoing
#-4:encoder control test
#-3:lidar servoing and grasping
#-2:yaw correction    #-1:navi to block test
# 0:navi to slot test # 1:lidar servoing   #2:line_tracking
# 3:apes 2h2          # 4:apes 4h2         #5:apes 6h2
# 6:apes 8h2          # 7:apes 8h2 v2 for 10
test_code: 5

# visual_line_tracking
localizer_Kp: 0.24
localizer_Kd: 0.16
localizer_speed: 80   
lr_diff_thresh: 10.0
lr_center_thresh: 400
lr_stop_thresh: 3
bot_stop_thresh: 3

# lidar_block_servoing
lidar_position_Kp: 140.0
lidar_position_Kd: 20.0
lidar_position_x_min_error: 0.015
lidar_position_y_min_error: 0.30
lidar_blind_forward_time: 1.3
lidar_forward_time_Kp: 0.6

# visual_block_servoing
position_Kp: 200
position_Kd: 10
position_x_min_error: 0.05
orientation_Kp: 0
orientation_up_start: 0.2
orientation_low_start: -0.2
stop_dist: 0
forward_time_Kp: 10

### cv_camera line_detection
device_id: 1
u_thresh: 110
hsv_b_low: 100
hsv_b_up: 140
maxArea: 2000
add_b_channel: false
b_thresh: 160
g_thresh: 180

### global_localizer line_tracker
left_x: 100     # 100 left bound of smaple region
num_x: 5        # 5  sample num in x direction
top_y: 50       # 50 top bound of smaple region
num_y: 3        # 3 sample num in y direction
bar_width: 30   # 30 width of sampled region
gap_width: 10   # 20 sample interval
count_thresh: 2 # 1-5 is okay only control y-axis
slider_area_low: 600    # 600 low silder area
slider_area_up: 1400    # 1400 up silder area low+800

### encoder 
left_encoder_Kp_fb: 2086.8    # encoder_fb_value = left_encoder_Kp_fb * distance
left_encoder_Kp_lr: 2242.6    # encoder_lr_value = left_encoder_Kp_lr * distance
right_encoder_Kp_fb: 0.0   # encoder_fb_value = right_encoder_Kp_fb * distance
right_encoder_Kp_lr: 0.0   # encoder_lr_value = right_encoder_Kp_lr * distance

